fedimint-sled/src/lib.rs

```diff
@@ -235,6 +243,18 @@ impl<'a> IDatabaseTransaction<'a> for SledTransaction<'a> {
         self.db.inner().flush().expect("DB failure");
         ret
     }
+
+    fn rollback_tx_to_savepoint(&mut self) {
+        // Remove any pending operations beyond the savepoint
+        let removed_ops = self.num_pending_operations - self.num_savepoint_operations;
+        for _i in 0..removed_ops {
+            self.operations.pop();
```

Comment:
Do I get it right that sled's implementation is kind of like a in-memory implementation and it doesn't really implement snapshot level isolation correctly?

Reply:
Yes that is correct. I have added a comment to clarify.